### PR TITLE
feat: add service account impersonation for access tokens

### PR DESCRIPTION
**Pull request by bshaffer**
Same as https://github.com/googleapis/google-auth-library-php/pull/580, but with all ID token logic removed (for now)

### CODE REVIEW

**Comment by bshaffer** [comment]
This would be great to get merged next (by Nov 15)

File: src/FetchAuthTokenInterface.php [hunk 1]
```diff
@@ -26,6 +26,7 @@ interface FetchAuthTokenInterface
      * Fetches the auth tokens based on the current state.
      *
      * @param callable|null $httpHandler callback which delivers psr7 request
+     *
```
Comment (Hectorhammett) [inline src/FetchAuthTokenInterface.php:29]: Nit, Is the @param and @return docs separated by a new line?

**Review by Hectorhammett — APPROVED** [review]

File: src/FetchAuthTokenInterface.php (hunk 1 above)
Comment (bshaffer) [reply src/FetchAuthTokenInterface.php:29]: Using regex, I found 8 instances where we had a newline between `@param` and `@return`, and 92 occurrences where we didn't.

Comment (bshaffer) [reply src/FetchAuthTokenInterface.php:29]:
```suggestion
```

**Pull request merged**